code/game/objects/items/storage/reagent_pouch.dm

```diff
@@ -0,0 +1,167 @@
+	inner.reagents.add_reagent(/datum/reagent/medicine/tricordrazine, amount_to_fill)
+	if(contents.len > 0)
+		var/obj/item/reagent_containers/hypospray/advanced/hypo_to_fill = contents[1]
+		amount_to_fill = hypo_to_fill.volume/4
+		hypo_to_fill.reagents.add_reagent(/datum/reagent/medicine/bicaridine, amount_to_fill)
+		hypo_to_fill.reagents.add_reagent(/datum/reagent/medicine/kelotane, amount_to_fill)
+		hypo_to_fill.reagents.add_reagent(/datum/reagent/medicine/tramadol, amount_to_fill)
+		hypo_to_fill.reagents.add_reagent(/datum/reagent/medicine/tricordrazine, amount_to_fill)
+		hypo_to_fill.update_icon()
+	update_icon()
+
+/obj/item/storage/pouch/pressurized_reagent_pouch/AltClick(mob/user)
+	if(!remove_canister(user))
+		return ..()
+
+///Removes the reagent canister from the pouch. Returns FALSE if there is no canister to remove
+/obj/item/storage/pouch/pressurized_reagent_pouch/proc/remove_canister(mob/user)
+	if(!inner)
+		to_chat(user, span_warning("There is no container inside this pouch!"))
+		return FALSE
+	if(!user.put_in_active_hand(inner))
+		user.put_in_hands(inner) //If put_in_active fails, we still pick up or drop the canister
+	inner = null
+	update_icon()
+	return TRUE
+
+/obj/item/storage/pouch/pressurized_reagent_pouch/attackby(obj/item/held_item, mob/user)
+	if(istype(held_item, /obj/item/reagent_containers/hypospray))
+		fill_autoinjector(held_item, user)
+		return ..()
+
+	if(istype(held_item, /obj/item/reagent_containers/glass/reagent_canister)) //If it's the reagent canister, we put it in the special holder
+		if(!inner)
+			user.temporarilyRemoveItemFromInventory(held_item)
+			inner = held_item
+			to_chat(user, span_notice("You insert [held_item] into [src]!"))
+			update_icon()
+			return
+		return to_chat(user, span_warning("There already is a container inside [src]!"))
+
+	return ..()
+
+/obj/item/storage/pouch/pressurized_reagent_pouch/attackby_alternate(obj/item/held_item, mob/user, params)
+	. = ..()
+	if(istype(held_item, /obj/item/reagent_containers/hypospray))
+		fill_autoinjector(held_item, user)
+
+///Fills the hypo that gets stored in the pouch from the internal storage tank. Returns FALSE if you fail to refill your injector
+/obj/item/storage/pouch/pressurized_reagent_pouch/proc/fill_autoinjector(obj/item/reagent_containers/hypospray/autoinjector, mob/user)
+	if(!inner)
+		user.balloon_alert(user, "No container")
+		return FALSE
+	if(inner.reagents.total_volume == 0)
+		user.balloon_alert(user, "No reagent left")
+		return FALSE
+	if(inner.reagents.total_volume > 0)
+		inner.reagents.trans_to(autoinjector, autoinjector.volume)
+		playsound(loc, 'sound/effects/refill.ogg', 25, TRUE, 3)
+		autoinjector.update_icon()
+		update_icon()
+
+/obj/item/storage/pouch/pressurized_reagent_pouch/examine(mob/user)
+	. = ..()
+	. += get_display_contents(user)
+
+///Used on examine for properly skilled people to see contents.
+/obj/item/storage/pouch/pressurized_reagent_pouch/proc/get_display_contents(mob/user)
```

Comment:
Also a thing about this proc, you're returning notices in some of the cases, while not in others. Why? Same thing applies for the other get_display... procs